app.rb

```diff
@@ -0,0 +1,82 @@
+require 'sinatra'
+require_relative 'lib/hangman_rules'
+require_relative 'lib/word_generator'
+require_relative 'lib/words'
+require_relative 'lib/word_formatter'
+
+class App < Sinatra::Base
+
+  enable :sessions
+
+  helpers do
+
+    def rules
+      HangmanRules.new(secret_word, guesses)
+    end
+
+    def letters
+      if session[:letters].nil?
+	session[:letters] = ("a".."z").to_a
+      end
+      session[:letters]
+    end
+
+    def secret_word
+      if session[:secret_word].nil?
+	session[:secret_word] = WordGenerator.new(Words.new).random_word
+      end
+      session[:secret_word]
+    end
+
+    def guesses
+      if session[:guesses].nil?
+	session[:guesses] = {:right_guesses => [], :wrong_guesses => []}
+      end
+      session[:guesses]
+    end
+
+    def images
```

Comment:
I think all this logic related to finding the image could be in its own class. And then you pass in the number of wrong guesses and it returns you the correct image.

Reply:
Yes, similar to how you did WordGenerator/Word